data/types_test.go

```diff
@@ -23,23 +23,29 @@ type TypesSuite struct{}
 
 var _ = Suite(&TypesSuite{})
 
+type ed25519Public struct {
+	PublicKey HexBytes `json:"public"`
+}
+
 func (TypesSuite) TestKeyIDs(c *C) {
 	var hexbytes HexBytes
 	err := json.Unmarshal([]byte(public), &hexbytes)
 	c.Assert(err, IsNil)
+	keyValBytes, err := json.Marshal(ed25519Public{PublicKey: hexbytes})
+	c.Assert(err, IsNil)
 
-	key := Key{
+	key := PublicKey{
 		Type:   KeyTypeEd25519,
 		Scheme: KeySchemeEd25519,
-		Value:  KeyValue{Public: hexbytes},
+		Value:  keyValBytes,
 	}
 	c.Assert(key.IDs(), DeepEquals, []string{keyid10})
```

Comment:
Can you also please test a case that ids is an array longer than 1.

Reply:
I think we should clean this up per https://github.com/theupdateframework/go-tuf/pull/148#discussion_r717629525
Our `IDs` function always returns a single value, it's hardcoded in `data.PublicKey` to do that.